library/common.h

```diff
@@ -334,4 +334,23 @@ static inline void mbedtls_xor_no_simd(unsigned char *r,
 #define MBEDTLS_OPTIMIZE_FOR_PERFORMANCE
 #endif
 
+/* Define macros that can be used to disable warnings about unreachable code. */
+#if defined(__clang__)
+
+#define MBEDTLS_PRAGMA(x) _Pragma(#x)
+
+#define MBEDTLS_IGNORE_UNREACHABLE_BEGIN \
+    MBEDTLS_PRAGMA(clang diagnostic push) \
+    MBEDTLS_PRAGMA(clang diagnostic ignored "-Wunreachable-code")
+
+#define MBEDTLS_IGNORE_UNREACHABLE_END \
+    MBEDTLS_PRAGMA(clang diagnostic pop)
+
+#else
+
+#define MBEDTLS_IGNORE_UNREACHABLE_BEGIN
```

Comment:
Styling is problematic with macros like these. Our version/configuration of uncrustify indents them in a sensible (aligned with the surrounding block and without affecting the indentation of surrounding code), but IDEs are likely to come up with something different that's also sensible. In practice we can typically write `MBEDTLS_IGNORE_UNREACHABLE_BEGIN;` and `MBEDTLS_IGNORE_UNREACHABLE_END;` on their own line to make the indentation naturally sensible.